conbench/config.py

```diff
@@ -87,10 +50,83 @@ class Config:
+        and ends with a slash.
+
+        Note: might want to do a little more URL-specific input validation via e.g.
+        https://validators.readthedocs.io/en/latest/#module-validators.url
+
+
+        `INTENDED_BASE_URL` is the base URL (scheme, DNS name, path prefix) that
+        regular HTTP clients are expected to use for reaching the HTTP server
+        exposing the app. Depends on the deployment and cannot generally be
+        determined by the app itself (requires human input). Meaningful values may
+        look like
+
+        https://conbench.ursa.dev/  or  http://127.0.0.1:9000/
+
+        This value is required to be provided by operators. That reduces code
+        complexity and enhances maintainability and security. Notably, this
+        simplifies code, reasoning, testing in the context of single sign-on.
+        """
+        ibu = os.environ.get("CONBENCH_INTENDED_BASE_URL", None)
+
+        if ibu is None:
+            sys.exit("enviroment: CONBENCH_INTENDED_BASE_URL is required but not set")
+
+        # Strip leading and trailing whitespace.
+        ibu = ibu.strip()
+
+        if not ibu.startswith("http"):
+            sys.exit(
+                f"CONBENCH_INTENDED_BASE_URL must start with 'http'. Got instead:`{ibu}`"
+            )
+
+        # Require trailing slash towards tidy URL generation.
+        if not ibu.endswith("/"):
+            ibu += "/"
+
+        return ibu
+
+    def _get_oidc_issuer_url_from_env_or_exit(self) -> str:
+        """Return `None` or a string.
+
+        If `OIDC_ISSUER_URL` is after all `None`: disable OpenID Connect (OIDC)
+        single sign-on. If this is not `None` then it must be a valid OIDC issuer
+        notation (that is, a URL).
+        """
+
+        oiu = os.environ.get("CONBENCH_OIDC_ISSUER_URL", None)
+        if oiu is not None:
+            assert oiu.startswith("http")
+            # Remove all trailing slashes to make URL construction predictable.
+            # ALos, the canonical form of an OIDC issuer URL does not have
+            # trailing slashes.
+            oiu = oiu.rstrip("/")
+        else:
+            # legacy config support: when CONBENCH_OIDC_ISSUER_URL is set in the
+            # environment it takes precedence. If it is not set and if
+            # GOOGLE_CLIENT_ID is set in the environment then set issuer to
+            # "https://accounts.google.com"
+            if "GOOGLE_CLIENT_ID" in os.environ:
+                oiu = "https://accounts.google.com"
+
+        # Require client ID and client secret to be set if issuer is configured.
+        # Those three parameters are all required for an OIDC authorization code
+        # flow.
+        if oiu is not None:
+            assert self.OIDC_CLIENT_ID is not None
+            assert self.OIDC_CLIENT_SECRET is not None
+
+        return oiu
+
+
+Config = ConfigClass()
+# for legacy code that imports via `from ..config import TestConfig`
+TestConfig = Config
```

Comment:
Towards more pythonic naming, in the future the `Config` object should be renamed, reflecting that it's a config singleton object, globally available, ideally immutable after app startup. All capital letters to indicate a constant. I have chosen the old name `Config` here to not break all those modules that import this object.